src/services/resources/diagrams/list-by-workflow/index.ts

```diff
@@ -0,0 +1,10 @@
+import { api } from "services/api/diagrams";
+
+export async function listByWorkflow(id: string) {
+  try {
+    const { data } = await api.get(`/diagrams/workflow/${id}`);
+    return data;
+  } catch (error) {
+    console.error("Erro ao listar os diagramas por workflow id", error);
+  }
+}
```

Comment:
Renomear a função para listByWorkflowId e alterar o parametro para workflowId

```suggestion
export async function listByWorkflowId(workflowId: string) {
  try {
    const { data } = await api.get(`/diagrams/workflow/${workflowId}`);
    return data;
  } catch (error) {
    console.error("Erro ao listar os diagramas por workflow id", error);
  }
}
```